File: parish360-core/src/main/java/org/parish360/core/ceremonies/dto/OrdinationInfo.java
package org.parish360.core.ceremonies.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.parish360.core.common.dto.Address;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class OrdinationInfo {
    private String religiousOrder;
    private String seminaryName;
    private Address seminaryAddress;
    private String previousOrdinationType;
    private LocalDate previousOrdinationDate;
    private Address previousOrdinationPlace;
}

File: parish360_mobile/lib/features/ceremonies/domain/entities/ordination_info.dart
import 'package:parish360_mobile/core/common/entities/place.dart';

class OrdinationInfo {
  final String? religiousOrder;
  final String? seminaryName;
  final Place? seminaryAddress;
  final String? previousOrdinationType;
  final DateTime? previousOrdinationDate;
  final Place? previousOrdinationPlace;

  OrdinationInfo({
    this.religiousOrder,
    this.seminaryName,
    this.seminaryAddress,
    this.previousOrdinationType,
    this.previousOrdinationDate,
    this.previousOrdinationPlace,
  });

  factory OrdinationInfo.fromJson(Map<String, dynamic> json) {
    return OrdinationInfo(
      religiousOrder: json['religious_order'] as String?,
      seminaryName: json['seminary_name'] as String?,
      seminaryAddress: json['seminary_address'] != null
          ? Place.fromJson(json['seminary_address'])
          : null,
      previousOrdinationType: json['previous_ordination_type'] as String?,
      previousOrdinationDate: json['previous_ordination_date'] != null
          ? DateTime.parse(json['previous_ordination_date'] as String)
          : null,
      previousOrdinationPlace: json['previous_ordination_place'] != null
          ? Place.fromJson(json['previous_ordination_place'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'religious_order': religiousOrder,
      'seminary_name': seminaryName,
      'seminary_address': seminaryAddress?.toJson(),
      'previous_ordination_type': previousOrdinationType,
      'previous_ordination_date': previousOrdinationDate,
      'previous_ordination_place': previousOrdinationPlace?.toJson(),
    };
  }
}
